Documentation: slicing pipeline in kedro-viz (#2043)
kedro-org/kedro-viz: docs/source/slice_a_pipeline.md

# Slice a pipeline

Slicing a pipeline refers to creating a subset of a pipeline's nodes, which can help in focusing on specific parts of the pipeline. There are two primary ways to achieve this in Kedro:

> 1. **Programmatically with the Kedro CLI.** This method is ideal for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/0.19.5/nodes_and_pipelines/slice_a_pipeline.html).

Reviewer: ```suggestion
1. **Programmatically with the Kedro CLI.** This method is ideal for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the Kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/latest/nodes_and_pipelines/slice_a_pipeline.html).
```

Should this point to `latest` in the docs? I don't remember what the normal practice is.

Reviewer 2: I don't think we're being very consistent, but at least if we want to avoid hardcoding a specific version, we should use `/stable` instead of `/latest`

2. **Visually through Kedro-Viz:** This approach allows you to visually choose and slice pipeline nodes, which then generates a run command for executing the slice within your Kedro project.

## Benefits of Kedro-Viz slicing

- **Visual Representation:** View the relationships between nodes and identify which ones are part of your slice.
- **Immediate Command Generation:** Get a ready-to-use CLI command for executing the sliced pipeline.
- **Interactive Control:** Visually select and reset slices with a couple of clicks.

## Steps to slice in Kedro-Viz

Kedro-Viz offers a user-friendly visual interface for slicing pipelines. Follow these steps to use the slicing feature:

1. **Select elements in the flowchart:** In Kedro-Viz, choose two elements to set the boundaries for your slice:
   - Click on the first node you want to include.
   - Hold the Shift key and select the second node.
   

> ![](./images/slice_pipeline_full_view.gif)

Reviewer 2: The GIF might need re-recording in light of https://github.com/kedro-org/kedro-viz/pull/2036#issuecomment-2302327774

(Also can we make it a bit bigger?)

Author: hey yeah, I'll update the gif once I push my changes to address your comment 👍

2. **Highlighted selection:** The flowchart will highlight all nodes between the selected elements, and the corresponding nodes in the list on the left will also be highlighted.

3. **View the run command:** After selecting the nodes, Kedro-Viz generates a CLI command for the sliced pipeline. You can copy this command and use it directly in your Kedro project to run the slice.

> 4. **Slice the pipeline:** When you're ready, click the "Slice" button. This opens a new view where you can directly interact with the sliced pipeline.

Reviewer 2: Maybe we can have another GIF with this action + reset?

# Slice a pipeline

> Slicing a pipeline refers to creating a subset of a pipeline's nodes, which can help in focusing on specific parts of the pipeline. There are two primary ways to achieve this in Kedro:

Reviewer 3: ```suggestion
Slicing a pipeline in Kedro refers to creating a subset of a pipeline's nodes, which can help in focusing on specific parts of the pipeline. There are two primary ways to achieve this:
```

> 1. **Programmatically with the Kedro CLI.** This method is ideal for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/stable/nodes_and_pipelines/slice_a_pipeline.html).

Reviewer 3: ```suggestion
1. **Programmatically with the Kedro CLI.** This method is suitable for those comfortable with command-line tools. Detailed steps on how to achieve this are available in the kedro documentation: [Slice a Pipeline](https://docs.kedro.org/en/stable/nodes_and_pipelines/slice_a_pipeline.html).
```

> 2. **Visually through Kedro-Viz:** This approach allows you to visually choose and slice pipeline nodes, which then generates a run command for executing the slice within your Kedro project.

Reviewer 3: ```suggestion
2. **Visually through Kedro-Viz:** This approach allows you to visually select and slice pipeline nodes, which then generates a run command for executing the slice within your Kedro project.
```

## Benefits of Kedro-Viz slicing

- **Visual Representation:** View the relationships between nodes and identify which ones are part of your slice.
- **Immediate Command Generation:** Get a ready-to-use CLI command for executing the sliced pipeline.
- **Interactive Control:** Visually select and reset slices with a couple of clicks.

## Steps to slice in Kedro-Viz

Kedro-Viz offers a user-friendly visual interface for slicing pipelines. Follow these steps to use the slicing feature:

> 1. **Select elements in the flowchart:** In Kedro-Viz, choose two elements to set the boundaries for your slice:

Reviewer 3: ```suggestion
1. **Select elements in the flowchart:** In Kedro-Viz, select two elements to set the boundaries for your slice:
```
